activerecord/lib/active_record/transactions.rb

```diff
@@ -330,6 +330,7 @@ def before_committed! # :nodoc:
     # Ensure that it is not called if the object was never persisted (failed create),
     # but call it after the commit of a destroyed object.
     def committed!(should_run_callbacks: true) #:nodoc:
+      @_start_transaction_state = { create: transaction_record_state(:new_record) }
```

Comment:
`committed!` and rolledback! pair are suppose to run callbacks and restore state only. I wonder if now we are adding more functionality to them that is needed.